### PR TITLE
Docs structure

### PR DESCRIPTION
**Pull request by grossbart**
This is a relatively large PR that is intended to be approved commit-by-commit instead of as a whole. I chose this structure because these things all belong together and depend on each other, so submitting multiple PRs did not really work.

You can find the current version of the documentation that is based on this PR at https://grossbart.github.io/fp-ts/ – this should make it easier to play around with the suggested documentation structure and provide feedback on what is good and what is not.

The documentation contains some dead ends (e.g. the tutorial sections have a tiny bit of content telling readers that they are under construction). This seems to be acceptable to me at this point because the whole documentation is still relatively fresh and not yet widely known; please object if you have a different opinion!

In general, I'm very open to changing things! I wrote, for example, the "core concepts" section but am not an expert, so I might have misused some technical terms or overly generalized. Also, if you have other ideas about the general structure, please let me know.

The documentation structure takes ideas from:

- https://www.divio.com/blog/documentation/
- https://redux.js.org/introduction/getting-started
- https://guides.rubyonrails.org/
- https://reactjs.org/docs/getting-started.html
- https://docs.djangoproject.com/en/2.1/

### CODE REVIEW

**Comment by gcanti** [comment]
@grossbart This is awesome. I love what you've done here, thanks so much!

File: docs/basics/index.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+title: Basic Tutorial
+permalink: /basics/
+has_children: true
+nav_order: 2
+has_toc: false
+---
+
+# Basic Tutorial
+
+The basic tutorial is a work in progress. It will be directed towards beginners once its ready.
```
Comment (richdouglasevans) [inline docs/basics/index.md]: `its` ➡️ `it's`

File: docs/advanced/index.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+title: Advanced Tutorial
+permalink: /advanced/
+has_children: true
+nav_order: 3
+has_toc: false
+---
+
+# Advanced Tutorial
+
+The advanced tutorial is a work in progress. It will be directed towards developers who are comfortable in using fp-ts and want to take it furher.
```
Comment (richdouglasevans) [inline docs/advanced/index.md]: `furher` ➡️ `further`

File: docs/introduction/core-concepts.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import { option, apply } from 'fp-ts'
+
+// liftA2 is an ad hoc polymorphic function and requires an Option instance.
+// This is what option.option provides.
+const liftA2Option = apply.liftA2(option.option)
+```
+
+Here's a small list of some commonly used data types, you can find more in the [API reference](../modules/).
+
+* [Option](../modules/Option.ts) – is used to explicitely express that a value can be absent.
```
Comment (richdouglasevans) [inline docs/introduction/core-concepts.md]: `explicitely` ➡️ `explicitly`

File: docs/advanced/index.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+{: .fs-6 .fw-300 }
+
+In the meantime, check out the [learning resources](../introduction/learning-resources) and the following resources:
+
+- "`fp-ts` to the max" (TypeScript port of John De Goes's ["FP to the max"](https://www.youtube.com/watch?v=sxudIMiOo68) in Scala)
+  - [Part I](https://github.com/gcanti/fp-ts/blob/master/tutorials/fp-ts-to-the-max-I.ts)
+  - [Part II](https://github.com/gcanti/fp-ts/blob/master/tutorials/fp-ts-to-the-max-II.ts)
+- [Free monad and `fp-ts` (code)](https://github.com/gcanti/fp-ts/blob/master/tutorials/Free.ts)
+- [MTL-style in `fp-ts` (code)](https://github.com/gcanti/fp-ts/blob/master/examples/mtl.ts)
```
Comment (richdouglasevans) [inline docs/advanced/index.md]: This link results in a _Page not found._

File: docs/advanced/index.md (hunk 2 above)
Comment (richdouglasevans) [reply docs/advanced/index.md]: Surround the mention of `fp-ts` in backticks for consistency.

File: docs/basics/index.md (hunk 1 above)
Comment (grossbart) [reply docs/basics/index.md]: Thanks, it's fixed! And no worries about being That Guy: there are whole job profiles who do nothing else than fix typos, it's important 😄

**Comment by grossbart** [comment]
Because this PR moves around several files and changes the structure of some content, I will merge this now so that we can use the new documentation structure as a baseline from here on out.

It is clear that there is still a lot of work ahead; I have a long list of things I would like to document and I tested the new documentation with some team mates today and it's starting to work, but they identified many smaller and larger things that can still be clarified. 

I would like to tackle this as an ongoing effort with smaller PRs from here on out instead of adding to this PR so as to prevent this from getting out of sync with master. Further feedback is always welcome, maybe it would be easiest to do this through issues with the "docs" label: https://github.com/gcanti/fp-ts/labels/docs

**Pull request merged**